### ISSUE TITLE
Wrong linestyle in legend with errorbars in PGFPlotsX

### ISSUE THREAD

**Issue by theomb21** [JuliaPlots/Plots.jl#2480]
Legend symbols are not correctly displayed in the legend if a series is added to a subplot where one series has error bars. This happens also if the series with error bars is a scatter-type.
MWE:
```
pgfplotsx()

## first series has error bars

# lines
plot(rand(10), yerr=0.1.*rand(10), label="solid line")# series is displayed with the correct legend symbol
plot!(rand(10), ls=:dash, label="dashed line")# series legend symbol is not dashed and does not match its color

# scatter
scatter(rand(10), yerr=0.1.*rand(10), shape=:rect, label="rect")# correct legend symbol
scatter!(rand(10), shape=:utriangle, label="utriangle")# wrong legend symbol
```

The Problem can be avoided by plotting first the series without error bars and then adding the series with error bars.
```
## add series with error bars
# lines
plot(rand(10), ls=:dash, label="dashed line")
plot!(rand(10), yerr=0.1.*rand(10), label="solid line")
# both legend symbols appear to be correct
```
However, if _two or more series with_ error bars are desired, it seems impossible to get the right legend.

```
## two series with error bars
# lines
plot(rand(10), yerr=0.1.*rand(10), label="solid line")# correct
plot!(rand(10), yerr=0.1.*rand(10), ls=:dash, label="dashed line")# wrong
```
I'm using Julia v1.3.1, PGFPlotsX v1.2.5 and Plots v0.29.7.

**Comment by BeastyBlacksmith** [comment]
can you try the master branch?

**Comment by BeastyBlacksmith** [comment]
Should be fixed in 0.29.8
